src/middlewared/middlewared/schema/attribute.py

```diff
@@ -12,7 +12,7 @@ class Attribute:
 
     def __init__(
         self, name='', title=None, description=None, required=False, null=False, empty=True, private=False,
-        validators=None, register=False, hidden=False, editable=True, example=None, **kwargs
+        validators:list[function]=None, register=False, hidden=False, editable=True, example=None, **kwargs
```

Comment:
This would be syntax error i.e `function` is not defined anywhere so this won't execute and the correct way to do this is with `typing.List`

Reply:
Also i think we should have a base validators class with `__call__` method raising `NotImplementedError` and all validators inheriting from it and we can specify that validator here like this
`typing.List[Validator]`

Reply:
I wasn't able to find a general Validator type, so I left it as List[Callable] for now.

Reply:
Okay so to clarify, can we add something like this
```
class ValidatorBase:
    def __call__(self, *args, **kwargs):
        raise NotImplementedError()


class Exact(ValidatorBase):
    def __init__(self, value):
        self.value = value

    def __call__(self, value):
        if value != self.value:
            raise ValueError(f"Should be {self.value!r}")
```
We can import the validator base here and then use it in type hints

Reply:
Thanks for the thorough review, I've updated the PR with everything you pointed out.